src/odemis/model/_futures.py

```diff
@@ -509,35 +509,56 @@ def __init__(self, futures):
         start = time.time()  # start time of the batch future
         super().__init__(start=start, end=start + sum(self.futures.values()))
         self.set_running_or_notify_cancel()
+        self.task_canceller = self._cancel_all  # takes care of cancelling the task (=all sub-futures)
 
         for f in self.futures:
-            f.add_update_callback(self._on_future_update)  # called whenever set_progress of any single future is called
-            f.add_done_callback(self._on_future_done)  # called when future is done
+            f.add_update_callback(self._on_future_update)  # called whenever set_progress of a sub-future is called
+            f.add_done_callback(self._on_future_done)  # called when a sub-future is done
 
     def _on_future_update(self, f, start, end):
         """
-        Whenever progress on the single future is reported, the progress (or end) for the batch future
+        Whenever progress on the single sub-future is reported, the progress (or end) for the batch future
         is updated accordingly.
 
+        :param f: (ProgressiveFuture) A single sub-future.
         :param start: (float) Start time of a single future in the batch future in seconds.
```

Comment:
Shouldn't this be:

> :param start: (float) Start time of a single future in the batch future in seconds since the epoch.

Reply:
I can add it here. There are more places in the futures module where this is not clarified. If I am looking at the code it looks like that the whole progress update functionality is only added in the progessive future. And there everything seems to depend on start/end values that are with respect to the epoch. But maybe I read this incorrectly.